### ISSUE TITLE
BatchNorm when coming from pytorch does not work.

### ISSUE THREAD

**Issue by Narsil** [Microsoft/onnxjs#30]
Simple Pytorch script to showcase the failure

```
#!/usr/bin/env python                                                            
import torch                                                                     
from torch import nn                                                             
                                                                                 
                                                                                 
def main():                                                                      
    model = nn.Sequential(nn.Conv2d(3, 8, kernel_size=3), nn.BatchNorm2d(8))     
    model.eval()                                                                 
                                                                                 
    image = torch.rand(1, 3, 224, 224)                                           
    torch.onnx.export(model, image, "out.onnx", verbose=True)                    
                                                                                 
                                                                                 
if __name__ == "__main__":                                                       
    main() 
```

When I try to load from the default script I get
`Scaler tensor is not implemented yet`.
Backtracking the issue, I think it is linked to Pytorch `BatchNorm.num_batches_tracked` tensor, which is a scalar.

By the way, why are scalars not supported yet ? Don't they behave as tensor of shape=[1] ? (broadcasting issues aside)

**Comment by fs-eire** [comment]
thanks for the feedback. we are working on a change to enable scaler tensor.

**Comment by Narsil** [comment]
I use the exact code from the README.
```
<html>
  <head>
  </head>

  <body>
    <!-- Load ONNX.js -->
    <script src="https://cdn.jsdelivr.net/npm/onnxjs/dist/onnx.min.js"></script>
    <!-- Code that consume ONNX.js -->
    <script>
      // create a session
      const myOnnxSession = new onnx.InferenceSession();
      // load the ONNX model file
      myOnnxSession.loadModel("./my-model.onnx").then(()=>{    // <<< Replace with out.onnx
        // generate model input
        const inferenceInputs = getInputs();
        // execute the model
        session.run(inferenceInputs).then(output=>{
          // consume the output
          const outputTensor = output.values().next().value;
          console.log(`model output tensor: ${outputTensor.data}.`);
        });
      })
    </script>
  </body>
</html>
```

And the script fails before loading the model (I tried removing the promise to be sure too)

**Comment by Narsil** [comment]
Seems #38 works.

**Comment by hariharans29** [comment]
Thanks for the feedback @Narsil. Can you please check if similarly #40 works for you? It should have similar support as #38 but better backend design for scalar tensors.

**Comment by NTT123** [comment]
@Narsil  Hi, I've removed the statistical scalar `num_batches_tracked` from Batchnorm which turned out to work.


```
import torch                                                                     
from torch import nn                                                             
                                                                                 
                                                                                 
def main():                                                                      
    model = nn.Sequential(nn.Conv2d(3, 8, kernel_size=3), nn.BatchNorm2d(8))     
    model.eval()                                                                 
                                                                                 
    def remove_batch_stat(m):
        if isinstance(m, nn.BatchNorm2d):
            del m._buffers['num_batches_tracked']

    model = model.apply(remove_batch_stat)

    image = torch.rand(1, 3, 224, 224)                                           
    torch.onnx.export(model, image, "out.onnx", verbose=True)                    
                                                                                 
                                                                                 
if __name__ == "__main__":                                                       
    main() 
```

**Comment by Narsil** [comment]
Actually using the modified version of onnxjs works and is even better ! 
But thank your for the tips @NTT123

**Comment by hariharans29** [comment]
Closing this issue. Please reopen if needed.
